src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -108,40 +109,55 @@ private static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> cha
                                  (byte)'8', (byte)'9', (byte)'a', (byte)'b',
                                  (byte)'c', (byte)'d', (byte)'e', (byte)'f');
 
+
+            nuint lengthSubVector128 = (nuint)bytes.Length - (nuint)Vector128<int>.Count;
             do
             {
                 // Read 32bits from "bytes" span at "pos" offset
                 uint block = Unsafe.ReadUnaligned<uint>(
                     ref Unsafe.Add(ref MemoryMarshal.GetReference(bytes), pos));
 
+                // TODO: Remove once cross-platform Shuffle is landed
+                // https://github.com/dotnet/runtime/issues/63331
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                static Vector128<byte> Shuffle(Vector128<byte> value, Vector128<byte> mask) =>
+                    Ssse3.IsSupported ? Ssse3.Shuffle(value, mask) : AdvSimd.Arm64.VectorTableLookup(value, mask);
```

Comment:
I believe we need slightly more complex pattern we need to follow for corelib so R2R works as expected (although I notably don't recall if that's just for "opportunistic" ISAs for a given platform or more general for unsupported platforms as well).
```csharp
if (Ssse3.IsSupported)
{
    return Ssse3.Shuffle(value, mask);
}
else if (!AdvSimd.Arm64.IsSupported)
{
     ThrowPlatformNotSupportedException();
}

return AdvSimd.Arm64.VectorTableLookup(value, mask);
```

Reply:
Does this mean Vector128 can't be used in corelib?

Reply:
No. We just have an expectation that corelib methods directly contain a relevant `IsSupported` or `IsHardwareAccelerated` check and that it has "some" path for the fallback case (generally this is just `throwing`): https://github.com/dotnet/runtime/blob/03e0187a1a912f38e718c00614201dd20fa1bff1/docs/design/coreclr/botr/vectors-and-intrinsics.md#code-review-rules-for-code-written-in-systemprivatecorelibdll